src/xcode/ENA/ENA/Source/Scenes/Home/HomeViewController.swift

```diff
@@ -43,34 +52,25 @@ final class HomeViewController: UIViewController {
 				exposureManagerState: exposureManagerState,
 				enState: initialEnState,
 				risk: risk
-			))
+			), exposureSubmissionService: exposureSubmissionService)
 		navigationItem.largeTitleDisplayMode = .never
-		addToUpdatingSetIfNeeded(homeInteractor)
+		delegate.addToUpdatingSetIfNeeded(homeInteractor)
```

Comment:
Why is this `addToUpdatingSetIfNeeded` needed?

Reply:
This is used to add certain view controllers to list of objects that need an update of exposure notification setting when it's been updated. Those controllers are ```ExposureNotificationSettingViewController``` and ```SettingsViewController```. The update is triggered from ```ENStateHandler```.

I could argue this design (and I did actually, but arguments were on the other side :)), but this is again an old design that I just placed a bit differently here.